.github/workflows/r_package.yml

```diff
@@ -257,58 +256,84 @@ jobs:
           cat ./tests.log
           exit ${exit_code}
-  test-r-debian-clang:
-    name: r-package (debian, R-devel, clang-${{ matrix.clang-version }})
+  test-r-extra-checks:
+    name: r-package (${{ matrix.image }}, R-devel)
     timeout-minutes: 60
     strategy:
       fail-fast: false
       matrix:
-        # list of versions tested in CRAN "Additional Checks":
-        # https://cran.r-project.org/web/checks/check_issue_kinds.html
-        clang-version:
-          - 16
-          - 17
+        # references:
+        #   * CRAN "additional checks": https://cran.r-project.org/web/checks/check_issue_kinds.html
+        #   * images: https://r-hub.github.io/containers/containers.html
+        image:
+          - clang16
+          - clang17
+          - clang18
+          - clang19
+          - gcc14
+          - intel
     runs-on: ubuntu-latest
-    container: rhub/debian-clang-devel
-    env:
-      DEBIAN_FRONTEND: noninteractive
+    container: ghcr.io/r-hub/containers/${{ matrix.image }}:latest
     steps:
-      - name: Install Git before checkout
-        shell: bash
-        run: |
-          apt-get update --allow-releaseinfo-change
-          apt-get install --no-install-recommends -y git
-      - name: Trust git cloning LightGBM
-        run: |
-          git config --global --add safe.directory "${GITHUB_WORKSPACE}"
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
           fetch-depth: 5
           submodules: true
-      - name: install clang
+      - name: Install pandoc
+        uses: r-lib/actions/setup-pandoc@v2
+      - name: Install LaTeX
+        shell: bash
         run: |
-          ./.ci/install-clang-devel.sh ${{ matrix.clang-version }}
+          if type -f apt 2>&1 > /dev/null; then
+            apt-get update
+            apt-get install --no-install-recommends -y \
+                devscripts \
+                texinfo \
+                texlive-latex-extra \
+                texlive-latex-recommended \
+                texlive-fonts-recommended \
+                texlive-fonts-extra \
+                tidy \
+                qpdf
+          else
+            yum update -y
+            yum install -y \
+                devscripts \
+                qpdf \
+                texinfo \
+                texinfo-tex \
+                texlive-latex \
+                tidy
+          fi
       - name: Install packages and run tests
         shell: bash
         run: |
-          export PATH=/opt/R-devel/bin/:${PATH}
           Rscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'knitr', 'markdown', 'Matrix', 'RhpcBLASctl', 'testthat'), repos = 'https://cran.rstudio.com', Ncpus = parallel::detectCores())"
           sh build-cran-package.sh
-          R CMD check --as-cran --run-donttest lightgbm_*.tar.gz || exit 1
+          R CMD check --as-cran --run-donttest lightgbm_*.tar.gz || true
```

Comment:
Could you please elaborate on this particular change?

Reply:
`R CMD check` redirects the output of compiling the library + installing it to a separate file. To be able to always see those logs in CI, we `cat` out that file:

https://github.com/microsoft/LightGBM/blob/6662d557d576c87a8951fb0b87a12df2fd725e1c/.github/workflows/r_package.yml#L299-L307

This `|| exit 1` prevents those lines from ever being reached.

Setting it to `|| true` will allow them to run, and then we'd be relying on the check on NOTEs / WARNINGs / ERRORs to appropriately raise a non-0 exit code and fail the build.

Reply:
Thanks, got it! I guess that NOTEs also cause exit code to be non-zero, that's why we cannot rely on it anymore like here
https://github.com/microsoft/LightGBM/blob/6662d557d576c87a8951fb0b87a12df2fd725e1c/.ci/test_r_package.sh#L200-L214

Reply:
I'm really glad you asked about this, because I realized some of the assumptions being made in this existing code are not correct.

I tested today with R 4.3.3 and found that NOTEs and WARNINGs do not lead to a non-0 exit code for `R CMD check`. ERRORs do.

So I think we do still want something like this `||`, to be sure that you can see all the build logs (content of `00install.out`) in CI logs for a build where `R CMD check` found ERRORs.

To avoid duplicating that in multiple places, I just pushed commits moving that into a new script, `.ci/run-r-cmd-check.sh`. In that script, I made "allowed number of NOTEs" parameterizable, so we can continue not allowing any NOTEs in most CI jobs, but selectively allow 1 in these new jobs.